internal/generate/olm-catalog/csv_updaters.go

```diff
@@ -226,39 +228,86 @@ var _ csvUpdater = crds{}
 
 // apply updates csv's "owned" CRDDescriptions. "required" CRDDescriptions are
 // left as-is, since they are user-defined values.
-// apply will only make a new spec.customresourcedefinitions.owned element for
-// a type if an annotation is present on that type's declaration.
 func (us crds) apply(csv *olmapiv1alpha1.ClusterServiceVersion) error {
-	ownedCRDs := []olmapiv1alpha1.CRDDescription{}
+	ownedDescs := []olmapiv1alpha1.CRDDescription{}
+	descMap := map[registry.DefinitionKey]olmapiv1alpha1.CRDDescription{}
+	for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+		defKey := registry.DefinitionKey{
+			Name:    owned.Name,
+			Version: owned.Version,
+			Kind:    owned.Kind,
+		}
+		descMap[defKey] = owned
+	}
 	for _, u := range us {
 		crd := apiextv1beta1.CustomResourceDefinition{}
 		if err := yaml.Unmarshal(u, &crd); err != nil {
 			return err
 		}
 		for _, ver := range crd.Spec.Versions {
-			// Parse CRD descriptors from source code comments and annotations.
-			gvk := schema.GroupVersionKind{
-				Group:   crd.Spec.Group,
+			defKey := registry.DefinitionKey{
+				Name:    crd.GetName(),
 				Version: ver.Name,
 				Kind:    crd.Spec.Names.Kind,
 			}
-			newCRDDesc, err := descriptor.GetCRDDescriptionForGVK(scaffold.ApisDir, gvk)
-			if err != nil {
-				if goerrors.Is(err, descriptor.ErrAPIDirNotExist) {
-					log.Infof("Directory for API %s does not exist. Skipping CSV annotation parsing for API.", gvk)
-				} else if goerrors.Is(err, descriptor.ErrAPITypeNotFound) {
-					log.Infof("No kind type found for API %s. Skipping CSV annotation parsing for API.", gvk)
-				} else {
-					return fmt.Errorf("failed to set CRD descriptors for %s: %v", gvk, err)
-				}
-				continue
+			if owned, ownedExists := descMap[defKey]; ownedExists {
+				ownedDescs = append(ownedDescs, owned)
+			} else {
+				ownedDescs = append(ownedDescs, olmapiv1alpha1.CRDDescription{
+					Name:    defKey.Name,
+					Version: defKey.Version,
+					Kind:    defKey.Kind,
+				})
+			}
+		}
+	}
+	csv.Spec.CustomResourceDefinitions.Owned = ownedDescs
+	sort.Sort(descSorter(csv.Spec.CustomResourceDefinitions.Owned))
+	sort.Sort(descSorter(csv.Spec.CustomResourceDefinitions.Required))
+	return nil
+}
+
+func updateDescriptions(csv *olmapiv1alpha1.ClusterServiceVersion, searchDir string) error {
+	// TODO: Should this skip generating descriptions if API directory exists
+	// but project type can't be detected due to layout differences.
+	if opType := projutil.GetOperatorType(); opType != projutil.OperatorTypeGo {
```

Comment:
Currently is not supportable but is possible gen the CSV for Helm/Ansible as I understand that the goal is we support it as well. So,  I do not think that we should have this because of this and because we do not do this skip for non-SDK projects as well. In this way, why to do it for helm/ansible types?

Reply:
@camilamacedo86 You can test this out locally but this does not skip generating CSVs for Helm/Ansible types. This only skips generating the owned CRD descriptions in the CSV. 
This kept the existing behavior for `generate csv` where it will skip `updateDescriptions()` and won't parse `pkg/apis` for CSV annotations for helm/ansible types since they don't have CSV annotations yet.

But your comment did make me reconsider why I'm explicitly detecting the Go project type when all we care about is parsing CSV annotations from the API directory e.g `pkg/apis`. 
Since we already handle the case of skipping annotation parsing for that case below I''m going to remove this check.

Reply:
Removed this check: https://github.com/operator-framework/operator-sdk/pull/2511/commits/3b3d3e08f4935a0c63524e9e7dc510980e10ef3d